packages/react/src/components/Storage/FileUploader/FileUploader/FileUploader.tsx

```diff
@@ -280,13 +289,13 @@ export function FileUploader({
           fileState: !validExtension ? FileState.ERROR : FileState.INIT,
           fileErrors: validExtension
             ? undefined
-            : translate('Extension not allowed'),
+            : displayText.extensionNotAllowed,
         };
 
         setFileStatuses(newFileStatuses);
       };
     },
-    [fileStatuses, setFileStatuses]
+    [fileStatuses, setFileStatuses, displayText]
```

Comment:
Prefer the specific values used within a hook to be used in dep arrays where possible, e.g. `displayText.extensionNotAllowed`. That way if for some reason `displayText` is re-evaluated it won't trigger a recalculation here unless the actual value of  `displayText.extensionNotAllowed` updates